src/category_theory/monad/basic.lean

```diff
@@ -56,4 +56,100 @@ attribute [simp] comonad.left_counit comonad.right_counit
 notation `ε_` := comonad.ε
 notation `δ_` := comonad.δ
 
+/-- A morphisms of monads is a natural transformation compatible with η and μ. -/
+structure monad_hom (M N : C ⥤ C) [monad M] [monad N] extends nat_trans M N :=
+(app_η' : ∀ {X}, (η_ M).app X ≫ app X = (η_ N).app X . obviously)
+(app_μ' : ∀ {X}, (μ_ M).app X ≫ app X = (M.map (app X) ≫ app (N.obj X)) ≫ (μ_ N).app X . obviously)
+
+restate_axiom monad_hom.app_η'
+restate_axiom monad_hom.app_μ'
+attribute [simp, reassoc] monad_hom.app_η monad_hom.app_μ
+
+/-- A morphisms of comonads is a natural transformation compatible with η and μ. -/
+structure comonad_hom (M N : C ⥤ C) [comonad M] [comonad N] extends nat_trans M N :=
+(app_ε' : ∀ {X}, app X ≫ (ε_ N).app X = (ε_ M).app X . obviously)
+(app_δ' : ∀ {X}, app X ≫ (δ_ N).app X = (δ_ M).app X ≫ app (M.obj X) ≫ N.map (app X) . obviously)
+
+restate_axiom comonad_hom.app_ε'
+restate_axiom comonad_hom.app_δ'
+attribute [simp, reassoc] comonad_hom.app_ε comonad_hom.app_δ
+
+namespace monad_hom
+variables {M N L K : C ⥤ C} [monad M] [monad N] [monad L] [monad K]
+
+@[ext]
+theorem ext (f g : monad_hom M N) :
+  f.to_nat_trans = g.to_nat_trans → f = g := by {cases f, cases g, simp}
+
+variable (M)
+/-- The identity natural transformations is a morphism of monads. -/
+def ident : monad_hom M M :=
+{ app_η' := by tidy,
+  app_μ' := by tidy,
+  ..𝟙 M }
```

Comment:
```suggestion
{ ..𝟙 M }
```

Reply:
Done. I actually tried this before, but it didn't work, I think because of the same `{x} : ...` vs. ` ... : \forall {x}, ...` issue.